src/main/java/com/aws/greengrass/util/EncryptionUtils.java

```diff
@@ -149,4 +152,83 @@ private static byte[] join(byte[] byteArray1, byte[] byteArray2) {
         System.arraycopy(byteArray2, 0, bytes, byteArray1.length, byteArray2.length);
         return bytes;
     }
+
+    /**
+     * Converts key to a PEM string.
+     *
+     * @param keyType    encoding format type of the key
+     * @param encodedKey encoded key
+     * @return a PEM string
+     * @throws IOException IO Exception
+     */
+    public static String keyToPem(String keyType, byte[] encodedKey) throws IOException {
+        try (StringWriter str = new StringWriter();
+             PemWriter pemWriter = new PemWriter(str)) {
+            pemWriter.writeObject(keyType, encodedKey);
+            pemWriter.close(); // Need to explicitly close this as it is a buffered writer
+            return str.toString();
+        }
+    }
+
+    // A generic PEM writer, based on RFC 1421
+    public static class PemWriter extends BufferedWriter {
```

Comment:
add a comment and copyright notice that this is taken from bouncycastle. We need to give appropriate attribution